src/devices/telldus.c

```diff
@@ -0,0 +1,195 @@
+/* 
+ * Telldus Thermo- & Hygrometer (312623, F007TPH)
+ * Proove Fridge/Freezer thermometer (311433)
+ *
+ * A complete message is 49 bits:
+ *      1-bit starter bit
+ *      48-bit data packet 
+ *
+ * 48-bit data packet format:
+ *
+ * 00000000 AAAAABBB CCCCDDDD DDDDDDDD EEEEEEEE FFFFFFFF 
+ *
+ * The first byte is always 00000000.
+ * A -	product id: this one never changes. 10110 (22)
+ * B -	channel: [CH1: 111] [CH2: 110] [CH3: 101] [CH4: 100] [CH5: 011] [CH6: 010] [CH7: 001] [CH8: 000]. Flipping bits + 1 gives channel.
+ * C -	sensor id: for models that have two sensors (and two displays) these four bits vary from 1000 (inside sensor) to 0111 (outside sensor).
+ * D -	temperature in Celsius: all of the bits have to be flipped and then divide by 10. MSB is a sign bit.
+ * E - 	Humidity: flip bits and it gives humidity-%. The byte is 00000000 if there is no humidity sensor.
+ * F -	CRC: non-standard CRC-8, 8 bits. 
+ *		http://reveng.sourceforge.net
+ *		width=8  poly=0x31  init=0x2e  refin=false  refout=false  xorout=0x00  check=0x57  residue=0x00  name=(none)
+ *  
+ * Copyright (C) 2017 Mika Kuivalainen
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+#include "rtl_433.h"
+#include "util.h"
+
+#define BITLEN 				48
+#define PACKETLEN    		6
+#define STARTBYTE	  		0x00
+#define INIT_CRC			0x2e
+
+/** telldus_crc8_bit():
+ * Compute CRC-8 for a byte sequence, one bit at a time.
+ *
+ * This function was generated using crcany (https://github.com/madler/crcany)
+ * License information: "This code is under the zlib license, permitting free
+ * commercial use."
+ */
+static unsigned telldus_crc8_bit(unsigned crc, void const *mem, size_t len) {
```

Comment:
Best to use thecrc8 from util.h https://github.com/merbanan/rtl_433/blob/master/include/util.h#L49